### PR TITLE
feat: update KP to KMS and add a new test

### PR DESCRIPTION
**Pull request by maheshwarishikha**
### Description

Update the repository to use the term KMS instead of KP.
It includes changes in Readme, variable name/description, default example and adds a test for existing KMS.

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

This repository now supports KMS including Key Protect and HPCS. Key protect term in the code has been renamed to KMS. Repository name is also changed to `terraform-ibm-kms-key-ring` from `terraform-ibm-key-protect-key-ring`.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Review by shemau — APPROVED** [review]
The code looks good.

There might be a naming issue/sequence issue. I think (not totally sure) that the .github/settings.xml is intended to change the name of the repo.
If/when the repo name changes, there are additional changes in the badges (README.md) and tests (go.mod) to reflect the new name.

There are NO changes to input variable names (variables.tf) or outputs; so I am not convinced this is major/breaking change. The module will get renamed, so there will be changes to all code consuming this module, but our current process deems these changes as non-breaking, since as far as I can tell there is no risk to deleting/recreating resources as a result of this change.

**Comment by maheshwarishikha** [comment]
@shemau sure, converted to minor release. Thanks.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
-# Key Protect key ring module
+# KMS key ring module
 
 [![Stable (With Quality Checks)](https://img.shields.io/badge/Status-Stable%20(With%20quality%20checks)-green?style=plastic)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
 [![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-key-protect-key-ring?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key-ring/releases/latest)
```
Comment (ocofaigh) [inline README.md:6]: `terraform-ibm-key-protect-key-ring` need to be updated in this badge url

File: README.md [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
 
-This module creates a key ring to help organize keys in a Key Protect instance.
+This module creates a key ring to help organize keys in a KMS instance.
 For more information, about key management rings, see [creating key rings](https://cloud.ibm.com/docs/key-protect?topic=key-protect-grouping-keys#create-key-ring-api).
```
Comment (ocofaigh) [inline README.md:10]: I think we need to explicitly mention Key Protect and HPCS are supported. the link here only points to key protect docs

**Review by ocofaigh — CHANGES_REQUESTED** [review]
some minor comments

File: README.md (hunk 1 above)
Comment (maheshwarishikha) [reply README.md]: updated

File: README.md (hunk 2 above)
Comment (maheshwarishikha) [reply README.md]: Made the required changes.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
-[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-key-protect-key-ring?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key-ring/releases/latest)
+[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-kms-key-ring?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-kms-key-ring/releases/latest)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
 
-This module creates a key ring to help organize keys in a Key Protect instance.
-For more information, about key management rings, see [creating key rings](https://cloud.ibm.com/docs/key-protect?topic=key-protect-grouping-keys#create-key-ring-api).
+This module creates a key ring to help organize keys in a KMS instance. KMS can be Key Protect and Hyper Protect Crypto Services(HPCS) instance.
```
Comment (ocofaigh) [inline README.md:9]: change `and` -> `or` here:

> KMS can be Key Protect **or** Hyper Protect Crypto Services(HPCS) instance

Comment (maheshwarishikha) [reply README.md]: updated

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**